### PR TITLE
[common-artifacts] Support bool for random data

### PR DESCRIPTION
**Pull request by seanshpark**
This will enable to support bool type for random data.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #7224, from draft #7223

File: compiler/common-artifacts/src/TestDataGenerator.cpp [hunk 1]
```diff
@@ -65,6 +65,15 @@ template <typename T> void geneate_random_data(std::mt19937 &gen, void *data, ui
...
 
+template <> void geneate_random_data<bool>(std::mt19937 &gen, void *data, uint32_t size)
+{
+  std::normal_distribution<float> distrib(0, 2); // mean(0), stddev(2)
+  for (uint32_t i = 0; i < size; i++)
+  {
+    static_cast<bool *>(data)[i] = distrib(gen) >= 0 ? true : false;
+  }
+}
+
```
Comment (mhs4670go) [inline compiler/common-artifacts/src/TestDataGenerator.cpp:76]: I'm not sure but is it necessary? I think template covers this bool case.

Comment (seanshpark) [reply compiler/common-artifacts/src/TestDataGenerator.cpp:76]: as the random value rang would be -2 ~ 2, mostly would be not 0, that is mostly would be true...
logically it is OK but I wanted to give equal possibility...

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**